src/org/omegat/core/search/ReplaceMatch.java

```diff
@@ -0,0 +1,46 @@
+/**************************************************************************
+ OmegaT - Computer Assisted Translation (CAT) tool
+          with fuzzy matching, translation memory, keyword search,
+          glossaries, and translation leveraging into updated projects.
+
+ Copyright (C) 20107 Thomas Cordonnier
+               Home page: http://www.omegat.org/
+               Support center: http://groups.yahoo.com/group/OmegaT/
+
+ This file is part of OmegaT.
+
+ OmegaT is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ OmegaT is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ **************************************************************************/
+
+package org.omegat.core.search;
+
+/**
+ * Extends search match for regular expression replacement:
+ * we must store the text for replacement because it will differ from one target string to another
+ *
+ * @author Thomas Cordonnier (t_cordonnier@yahoo.fr)
+ */
+public class ReplaceMatch extends SearchMatch {
```

Comment:
In your other comments you don't say why. I wrote like this because we are in an object-oriented language, where it seems logical to consider that a replace match is a special case of search match which contains also text to replace with.
I prefer that rather than big containers like search expression, where you have to look for some fields to decide whenever another one should be filled or not (for example rootDir should be filled only if mode == search)
In case you wrote something about that in the coding guidelines, can you give a link?

An alternative could be the following:
class SearchMatch {
            ...
           class ReplaceMatch {
                    public String replacement;
                    public SearchMatch getSearch() { return SearchMatch.this; }
           }
}

Reply:
Yes, I meant because of the cast. Sorry my comments were so scattered; I'll gather my thoughts on this below, not mixed in with the code.